ydb/core/tx/datashard/export_s3_buffer.cpp

```diff
@@ -1,31 +1,138 @@
 #ifndef KIKIMR_DISABLE_S3_OPS
 
-#include "export_s3_buffer_raw.h"
+#include "export_s3_buffer.h"
 #include "type_serialization.h"
 
+#include <ydb/core/backup/common/checksum.h>
 #include <ydb/core/tablet_flat/flat_row_state.h>
 #include <yql/essentials/types/binary_json/read.h>
 #include <ydb/public/lib/scheme_types/scheme_type_id.h>
 
 #include <library/cpp/string_utils/quote/quote.h>
 
 #include <util/datetime/base.h>
+#include <util/generic/buffer.h>
 #include <util/stream/buffer.h>
 
-namespace NKikimr {
-namespace NDataShard {
+#include <contrib/libs/zstd/include/zstd.h>
 
-TS3BufferRaw::TS3BufferRaw(const TTagToColumn& columns, ui64 rowsLimit, ui64 bytesLimit, bool enableChecksums)
-    : Columns(columns)
-    , RowsLimit(rowsLimit)
-    , BytesLimit(bytesLimit)
-    , Rows(0)
-    , BytesRead(0)
-    , Checksum(enableChecksums ? NBackup::CreateChecksum() : nullptr)
+
+namespace NKikimr::NDataShard {
+
+namespace {
+
+struct DestroyZCtx {
+    static void Destroy(::ZSTD_CCtx* p) noexcept {
+        ZSTD_freeCCtx(p);
+    }
+};
+
+class TZStdCompressionProcessor {
+public:
+    using TPtr = std::unique_ptr<TZStdCompressionProcessor>;
```

Comment:
Для консистентности:
```suggestion
    using TPtr = THolder<TZStdCompressionProcessor>;
```

Чтобы потом можно было махом во всём файле поменять.